### DISCUSSION TITLE
Export to Cytoscape

### DISCUSSION THREAD

**Discussion by zljnu** [metgem/metgem#97] [Q&A] [answered]
Hey there,

How can I export the network to Cytoscape with the node and edge tables, in line with the imported metadata file. The default result missed the above tables in Cytoscape.

Many thanks!

**Comment by n-elie** [comment]
Hello,

You could use the 'Export metadata as CSV' function in MetGem and then import the resulting file in Cytoscape.

Regards,
Nicolas

**Reply by zljnu** [reply]
Hi Nicolas,

Thanks for you help. I can import the exported metadata in cytoscape following your suggestion, but the node table in table panel looks wired, the share name or id is different from those in the original exported metadata file (attached picture below). How can I normalized them so that the node information in the cytoscape network could be accurately correlated.

Best,
Zhenlong

![20211011162827](https://user-images.githubusercontent.com/92217619/136758186-ea661616-e27c-480f-b46a-d447de680b32.png)

**Reply by n-elie** [reply]
I am not sure that I understand well the problem. Did you use 'Import table from file" button in Cytoscape? Did you set the column 'id' as key like in this screenshot?
![image](https://user-images.githubusercontent.com/40382614/136779870-5631035c-437b-4344-9c85-463fe2e8408c.png)

**Comment by FlepTumbleweed** [comment]
Hi,

I had a similar problem. The issue might be, that MetGem exports the network nodes numbered from 1 to the last node. Other softwares (e.g. mzmine) from which I exported quantification metadata from, keep their original IDs (e.g. from before removing Ions due to threshold filtering or whatever filtering you do). So I ended up with 447 nodes and 447 metadata entries, but the network was numbered from 1 to 447, the Metadata from 1 to around 4000 (with gaps, e.g. [1, 2, 6, 14, ..., 4000]).

A **workaround solution** for you might be to manually number your metadata IDs to match the network nodes before importing. While MetGem seems to replace the original IDs, the order seems to stay identical (best try it yourself as well, I just highlighted the first 10 nodes in MetGem and Cytoscape for a test and they were identical).  

Of course it would be ideal if network export had an option to keep the original labels (although the problem might be caused by Cytoscape as well).

Best regards,
Flep

**Comment by n-elie** [comment, marked as answer]
Indeed, MetGem re-numbers the spectra at import (it simply reads them in the order they were exported). So, IDs of metadata should be reset before export using the 'Feature list methods > Filtering > Feature list rows filter' option in MZMine:
![image](https://user-images.githubusercontent.com/40382614/137924772-70148ed3-2d26-4155-ae63-6f205816bebb.png)
Nicolas

**Reply by FlepTumbleweed** [reply]
I guess that makes my workaround obsolete - thanks. 👍 

Flep
